apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java

```diff
@@ -1050,4 +1030,38 @@ public void addProviderSliceSetsOrganizationAndEobCorrectlyWithDefaultNpiValueOf
             .findFirst()
             .isPresent());
   }
+
+  /**
+   * Verifies that providing a EnumSet of {@link ClaimTypeV2} and a bit mask integer denoting claim
+   * types that have data, teh results is a filtered EnumSet.
+   */
+  @Test
+  public void verifyEnumSetFromListOfClaimTypesAndDatabaseBitmaskOfData() {
+    EnumSet<ClaimTypeV2> allClaimSet = EnumSet.allOf(ClaimTypeV2.class);
+
+    // resultant set only includes claim types that have data.
+    int testVal = QueryUtils.V_DME_HAS_DATA | QueryUtils.V_SNF_HAS_DATA | QueryUtils.V_HHA_HAS_DATA;
+    EnumSet<ClaimTypeV2> availSet =
+        TransformerUtilsV2.fetchClaimsAvailability(allClaimSet, testVal);
+
+    assertTrue(availSet.contains(ClaimTypeV2.HHA));
+    assertTrue(availSet.contains(ClaimTypeV2.SNF));
+    assertTrue(availSet.contains(ClaimTypeV2.DME));
+    assertFalse(availSet.contains(ClaimTypeV2.INPATIENT));
+
+    // check efficacy of EnumSet filter vs. bit mask of data.
+    EnumSet<ClaimTypeV2> someClaimSet = EnumSet.noneOf(ClaimTypeV2.class);
+    someClaimSet.add(ClaimTypeV2.CARRIER);
+    someClaimSet.add(ClaimTypeV2.PDE);
+
+    availSet = TransformerUtilsV2.fetchClaimsAvailability(someClaimSet, testVal);
+    assertFalse(availSet.contains(ClaimTypeV2.HHA));
+    assertFalse(availSet.contains(ClaimTypeV2.SNF));
+    assertFalse(availSet.contains(ClaimTypeV2.DME));
+    assertFalse(availSet.contains(ClaimTypeV2.CARRIER));
+    // adjust data bit mask and try again
+    testVal = testVal | QueryUtils.V_CARRIER_HAS_DATA;
+    availSet = TransformerUtilsV2.fetchClaimsAvailability(someClaimSet, testVal);
+    assertTrue(availSet.contains(ClaimTypeV2.CARRIER));
```

Comment:
assertj might make this more expressive.

https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#filters

Or, even easier, can't `assertEquals()` be used on the set instead of a sequence of `contains` calls?

Reply:
Actually, this test would be much more readable if all of the `assertFalse`s were replaced by a single `assertEquals(Set.of(...expected...), availSet);`